src/sinks/splunk_hec/common/mod.rs

```diff
@@ -4,4 +4,18 @@ pub mod response;
 pub mod service;
 pub mod util;
 
+use serde::{Deserialize, Serialize};
 pub use util::*;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "snake_case")]
```

Comment:
Just curious why rename the fields to snake_case?

Reply:
Good point. `lowercase` would make more sense. So we can do:

```toml
endpoint_target = "raw"
```

Reply:
We commonly rename enum variants to `snake_case` so that users configure options like `foo = "some_value"` and not `foo = "SomeValue"`. In this case `lowercase` is the same as `snake_case` since they are all single-world values, but I'd recommend sticking with `snake_case` for consistency in case we add another enum variant that is multiple words and forget to update the `rename_all` to `snake_case`.

Reply:
Ah ok, that's the background context I was looking for.